lib/src/main/kotlin/com/squareup/cash/kfsm/State.kt

```diff
@@ -33,8 +36,12 @@ open class State(transitionsFn: () -> Set<State>) {
 object StateMachine {
 
   /** Check your state machine covers all subtypes */
-  fun <S : State> verify(head: S, kClass: KClass<S>) = walkTree(head).flatMap { seen ->
-    val notSeen = kClass.sealedSubclasses.minus(seen.map { it::class }.toSet()).toList().sortedBy { it.simpleName }
+  @Suppress("UNCHECKED_CAST") fun <S : State> verify(head: S) = verify(head, head::class.allSuperclasses
```

Comment:
We can avoid the unchecked cast by using `reified` instead, to access the type at runtime after type erasure.

It does make the interface a bit more awkward however.

E.g.
```kotlin
inline fun <reified S : State> verify(head: S) = walkTree(head).flatMap { seen ->
  // This is notably different, as it fetches the superclass from the reified type
  val notSeen = S::class.superclasses.first().sealedSubclasses.minus(seen.map { it::class }.toSet()).toList().sortedBy { it.simpleName }
  if (notSeen.isEmpty()) {
    seen.right()
  } else {
    "Did not encounter [${notSeen.map { it.simpleName }.joinToString(", ")}]".left()
  }
}

fun walkTree(
  current: State,
  statesSeen: Set<State> = emptySet()
): Either<String, Set<State>> =
  when {
    statesSeen.contains(current) -> statesSeen.right()
    current.transitions.isEmpty() -> statesSeen.plus(current).right()
    else -> current.transitions.map { walkTree(it, statesSeen.plus(current)) }.reduce { a, b ->
      when {
        a.isLeft() -> a
        b.isLeft() -> b
        else -> a.flatMap { left -> b.map { right -> left.plus(right) } }
      }
    }
  }

```